pretrain/tinyllama.py

```diff
@@ -22,51 +22,41 @@
 from lightning.pytorch.loggers import WandbLogger
 from torch.utils.data import DataLoader
 from torchmetrics.aggregation import RunningMean
+from typing_extensions import Literal
 
 # support running without installing as a package
 wd = Path(__file__).parent.parent.resolve()
 sys.path.append(str(wd))
 
+from lit_gpt.args import EvalArgs, IOArgs, TrainArgs
 from lit_gpt.model import GPT, Block, CausalSelfAttention, Config, LLaMAMLP
 from lit_gpt.utils import CycleIterator, chunked_cross_entropy, num_parameters
 
-# System settings
-model_name = "tiny-llama-1.1b"
-name = "lit-tiny-llama-1.1b"
-out_dir = Path(os.getenv("LIGHTNING_ARTIFACTS_DIR", "out")) / name
-logger_name = "tensorboard"
-devices = torch.cuda.device_count() or 1
 
-# Hyperparameters
-global_batch_size = 512
-learning_rate = 4e-4
-micro_batch_size = 4
-max_tokens = int(3e12)  # 3 trillion
-warmup_steps = 2000
-log_step_interval = 1
-eval_iters = 100
-save_step_interval = 1000
-eval_step_interval = 1000
-
-weight_decay = 1e-1
-beta1 = 0.9
-beta2 = 0.95
-grad_clip = 1.0
-decay_lr = True
-min_lr = 4e-5
-
-batch_size = global_batch_size // devices
-gradient_accumulation_iters = batch_size // micro_batch_size
-assert gradient_accumulation_iters > 0
-warmup_iters = warmup_steps * gradient_accumulation_iters
-log_iter_interval = log_step_interval * gradient_accumulation_iters
```

Comment:
Where did this part go? It looks like these computations have simply disappeared.

Reply:
batch size and gradient accumulation iters are now dataclasses methods (see args.py)

log_iter_interval is in L182: https://github.com/Lightning-AI/lit-gpt/pull/913/files#diff-00fedfb359238295e8b95bc7c9f90975463bf4c40f03b2a010a118ec21f2f8d3R182